### PR TITLE
fix(spec): a wrapper's status parameter is not its response body

### LINKED ISSUE #485: An error helper's status argument can be resolved as the response body
Found while measuring #470 (PR #484), and split out because it is a mis-resolution in response extraction rather than anything to do with media types.

## What

On a large real project (gitea, 931 paths), **eighteen** error responses resolve a body whose Go type is `untyped int` — the **status argument** of an error helper, read as the payload:

```go
// services/context/base.go
func (b *Base) HTTPError(status int, contents ...string) {
	v := http.StatusText(status)
	if len(contents) > 0 {
		v = contents[0]
	}
	http.Error(b.Resp, v, status)
}

// routers/common/markup.go:94
ctx.HTTPError(http.StatusUnprocessableEntity, "unsupported render mode: "+mode)
```

The real body is `v`, the string, correctly documented as `text/plain; charset=utf-8` by the `http.Error` underneath. The second fragment describes `http.StatusUnprocessableEntity` — a literal — as an `application/json` body of `{type: integer}`.

Every one of the eighteen is the same shape:

```
POST 422 /-/markup                              application/json  {"type": "integer"}
POST 500 /-/markup                              application/json  {"type": "integer"}
GET  404 /org/{org}/worktime                    application/json  {"type": "integer"}
GET  404 /{username}                            application/json  {"type": "integer"}
… 14 more, all {"type": "integer"}
```

with `BodyType` = `untyped int` in every case.

## Why it has not been visible

A status held one body until #470, so one fragment per status displaced the other and the text/plain reading won. Nothing in the output showed the second reading existed. #470 makes a status able to carry several representations, which would have surfaced all eighteen as `application/json: {type: integer}` on error responses that send no such thing.

PR #484 therefore gates composition on the body being a value the handler **serialises**, excluding untyped constants. That is a stopgap in the upper layer and is described as one in the code: it stops the mis-resolution reaching the document, and does not fix it.

## Cost of the stopgap

It also rejects a genuine payload that happens to be a named untyped constant:

```go
const legacyCode = 7
func h(w http.ResponseWriter, r *http.Request) {
	if r.Header.Get("Accept") == "application/xml" {
		w.Header().Set("Content-Type", "application/xml")
		_ = xml.NewEncoder(w).Encode(legacyCode)
		return
	}
	_ = json.NewEncoder(w).Encode(legacyCode)
}
```
→ only the XML representation is documented. Raised by CodeRabbit on #484; rare, but real, and it is the reason this issue exists rather than the gate being called a fix.

(A string or numeric **literal** — `Encode("ok")` — is unaffected: `determineLiteralType` resolves those to `string`/`int`, not to an untyped form. Verified with a probe.)

## Not reproduced in a fixture yet

A minimal replica of the shape — a `Base.HTTPError(status int, contents ...string)` wrapper around `http.Error`, called with a status constant and a message — documents the 422 correctly as text/plain only. So something else in gitea's wiring is required: the variadic chain through `web.Bind`, a derived wrapper response pattern, or the depth at which the helper is reached. Worth finding before fixing, since the fix depends on which pattern is reading argument 0.

## Suggested direction

1. Reproduce it — start from the derived wrapper response patterns (`mergeResponseRoles` in `wrapper_detect.go`), which map an inner call's argument roles back onto a wrapper's parameters; a variadic parameter there is the same hazard #416 fixed for handlers.
2. With the cause known, the status argument should never be eligible as a body — the pattern that read it knows which index is which.
3. Then PR #484's `serialisedBody` gate can be removed, and the `legacyCode` shape above starts working.

## Acceptance

- A fixture reproducing the `untyped int` body, failing before the fix.
- The eighteen gitea entries stay absent with `serialisedBody` removed.
- The named-untyped-constant payload above documents both representations.

### PR DESCRIPTION
**Pull request by ehabterra**
Closes #485, and removes the stopgap #484 introduced.

## Root cause, found by instrumenting the derivation

```
callee=HTTPError nargs=2 typeIdx=0 statusIdx=0 callRegex="^HTTPError$"
```

Both indices point at argument 0 — the **status**.

```go
func (b *Base) HTTPError(status int, contents ...string) {
	v := http.StatusText(status)   // v traces back to `status`
	if len(contents) > 0 {
		v = contents[0]
	}
	http.Error(b.Resp, v, status)
}
```

The inner `http.Error` names its body at argument 1 and its status at 2, both correct. Mapped back onto `HTTPError`'s parameters, the status lands on `status` — right — and the body follows `v` through the first assignment it finds and lands on `status` too. The derived pattern then documents a status constant as the payload.

One parameter cannot be both, so the derivation declines the body. Guessing the other branch of that assignment would be a coin flip between `contents` and nothing (golden rule #7).

## The status half is kept — and my first attempt got this wrong

Declining the body initially dropped the whole pattern, because `Complete = TypeFromArg`. Measured, that cost **186 statuses** on gitea, every one of them correct. `ctx.HTTPError(404)` really does answer 404 with no body; saying so beats both inventing a body and dropping the status with it.

## Measurements (gitea)

| | |
|---|---|
| invented bodies removed | **214** |
| …of which `{type: integer}` (the status code itself) | 209 |
| …of which `[]byte` | 5 |
| response keys lost | **0** |
| paths lost | **0** |
| `$ref`s among the removals | **0** — no component-backed body touched |

All at error statuses: 403 (89), 500 (44), 401 (34), 404 (25), 400 (16), 422 (4), 409 (2).

Fixtures: **123 of 123 identical**.

## Removes #484's stopgap

That PR gated media-type composition on the body not being an untyped constant, purely to stop these bodies reaching the document. With the cause fixed the gate is gone, and the case it wrongly rejected works again — a genuine payload that happens to be a named untyped constant documents both its representations:

```go
const legacyCode = 7
// XML branch and JSON branch
→ {"application/json": {"type":"integer"}, "application/xml": {"type":"integer"}}   ✓ both
```

## Fixture

#485 recorded that a minimal replica did **not** reproduce this. It does — the missing ingredient was that the junk body is only visible once #470's composition is active, since before that the text/plain fragment displaced it. `testdata/wrapper_status_not_body` is forty lines, and its structural test fails without this change:

```
422 documents an application/json body — the only value at that status is the message
    http.Error writes as text/plain; a JSON body here is the status constant read as the payload
422 documents an integer body (application/json) — that is the status code itself
```

The unit test covers the index-0 trap: `StatusArgIndex` defaults to 0, so a pattern that does *not* read its status from an argument must not veto a body there — which is where most single-argument responders put it.

Suite, `make lint`, coverage ratchet (96.0%) pass.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved HTTP response documentation to prevent status values from being incorrectly documented as response bodies.
  - Preserved status-only error responses without inventing payload schemas.
  - Continued documenting valid response content types and bodies accurately.

- **Tests**
  - Added coverage for status-only responses and alternate response composition edge cases.
  - Added regression tests confirming valid response bodies remain documented correctly.

### CODE REVIEW

File: internal/spec/wrapper_detect.go [hunk 1]
```diff
@@ -664,19 +664,63 @@ func mergeResponseRoles(byMethod map[string]*DetectedWrapper, key string, w *wra
 			entry.Response.StatusFromArg, entry.Response.StatusArgIndex = true, i
 		}
 	}
+	statusOnly := false
 	if inner.TypeFromArg && !entry.Response.TypeFromArg {
 		if i, ok := params.indexOf(w, argAt(edge, inner.TypeArgIndex)); ok {
-			entry.Response.TypeFromArg, entry.Response.TypeArgIndex = true, i
-			entry.Response.Deref = inner.Deref
+			if sameAsStatusParam(entry.Response, i) {
```
[the remark this answers is not in the document]
Comment (ehabterra) [reply internal/spec/wrapper_detect.go]: Both halves confirmed and fixed in 393a83f — this is a real order-dependence bug in my own change, and the reasoning is exactly right.

**The collision check ran only while adding the body.** So a body-only edge reaching the merge first sets the body role with no status recorded yet — nothing to collide with — and a status-only edge afterwards lands on the same parameter, reproducing by the other route the precise state the check exists to prevent. `detectValueRound` walks the call graph in stored order, so the derivation does not get to assume which arrives first; I had effectively assumed it.

It is now a **normalisation over the roles accumulated so far**, run after either changes. The status is what survives a collision: it is read straight from an argument the caller passes, while the body reached the same parameter by following a local through an assignment — the weaker inference, and the one that was wrong in every observed case.

**Your second half is the same shape and I had missed it entirely.** `statusOnly` was a local recomputed per merge, so a later status-only edge cleared `Complete` for a responder deliberately kept for its status. Now a sticky field on the entry, set where the body is declined and read where completeness is decided — because the merge that discovers the collision is not the merge that finishes the pattern.

Both orders are pinned, plus the two cases that must **not** change: distinct parameters keep both roles, and a declined body leaves `TypeArgIndex` at -1 so a consumer reading it without checking `TypeFromArg` cannot land on the status by accident.

Re-measured after the change: gitea byte-identical to the pre-review build (214 invented bodies still removed, 0 statuses lost), 122 of 123 fixtures identical with the one difference being this PR's own new case.

File: internal/spec/extractor.go [hunk 2]
```diff
@@ -4099,43 +4099,7 @@ func otherMediaType(cur, next *ResponseInfo) bool {
...
-// encoded in two formats) is rejected too, and only one of its representations
-// is documented. A string or numeric LITERAL is unaffected — determineLiteralType
-// resolves `Encode("ok")` to `string`, not to an untyped form.
-func serialisedBody(r *ResponseInfo) bool {
-	if r == nil || r.BodyType == "" {
-		return false
-	}
-	_, untyped := untypedConstantDefault(r.BodyType)
-	return !untyped
+	return cur.BodyType != "" && next.BodyType != ""
```
[the remark this answers is not in the document]
Comment (ehabterra) [reply internal/spec/extractor.go:4102]: Agreed and added in 393a83f.

You are right that the gap was real: `renderer_content_types` exercised alternate media only with a struct, and the untyped-constant probe only ever used JSON — so the specific case this PR **recovers** had no test at the extraction level. A regression in `otherMediaType` would have gone unnoticed for precisely the shape #484's stopgap used to reject.

`testdata/renderer_content_types` now has a named untyped constant encoded in both formats:

```go
const legacyCode = 7
// XML branch and JSON branch
```
```
GET /items/legacy 200: {"application/json": {"type":"integer"}, "application/xml": {"type":"integer"}}
```

The test asserts both media types **and** the constant's default type on each side, so neither a dropped representation nor a schema invented for the constant can pass.
